Dockerfile.pulumi

```diff
@@ -1,4 +1,6 @@
-FROM python:3.8-slim-bullseye as grapl-local-pulumi
+FROM python:3.10-slim-bullseye as grapl-local-pulumi
```

Comment:
Once we've got everything on 3.10 --- that is, when our `.python-version` file contains a 3.10.x version --- then we could rig something up so that we pass that version in as a build argument and incorporate that into the base image identifier. That would be one less thing we'd have to keep in sync. (That goes for the `src/python/Dockerfile`, as well.)

(That's not a change for this PR, of course... just mentioning the idea for the future.)

Reply:
Prior art for Rust: https://github.com/grapl-security/grapl/pull/1788/commits/bdefc4d1ffeb5fade35ae203d04f0aa113feeec2

Reply:
Actually, let's just make an issue, shall we? 😅 https://github.com/grapl-security/issue-tracker/issues/973